internal/metrics/recorder.go

```diff
@@ -356,10 +359,14 @@ func (r *ReconcileError) Reason() string {
 
 // Wraps 2 errors and handles nil errors
 func (r *ReconcileError) Join(err1 error, err2 error) error {
-	if err1 == nil || err2 == nil {
+	if err1 == nil && err2 != nil {
+		return err2
+	} else if err2 == nil && err1 != nil {
 		return err1
+	} else if err1 == nil && err2 == nil {
+		return nil
```

Comment:
This can be converted to a neat switch statement.

Reply:
Thanks but I actually prefer explicitness which this if-else block describes it pretty well. If a switch statement shortens it, then I would not prefer that as the logic here is ver important, otherwise there would be no difference.

Reply:
I understand your point, but the switch case conveys the explicitness that is desired IMO, multi-if-else statements aren't usually considered a good practice, makes the codebase look sluggish if you get me.

Ideally, I'd like this function to look something like:

```
func (r *ReconcileError) Join(err1 error, err2 error) (err error) {
    switch {
    case err1 == nil && err2 != nil:
        err = err2
    case err1 != nil && err2 == nil:
        err = err1
   default:
        // do nothing or you can hard assign err = nil
    } 
	return fmt.Errorf("%v: ", err)
```

Reply:
There are cases where switch is better than an if statement, the same way the other way around and this usually depends on the proglang (compiler) in use and the constructs being compared. In this case, I don't really see much difference with respect to readability or speed, the conditions are pretty short. But if that's what it takes for this PR to be approved I will then update this so it can be cleared in our pipeline. I will revise.